mono/metadata/locales.c

```diff
@@ -515,7 +516,8 @@ get_darwin_locale (void)
 		CFRelease (locale);
 	}
 
-	return g_strdup (darwin_locale);
+	mono_memory_barrier ();
+	return g_strdup (cached_locale = darwin_locale);
```

Comment:
You would need to us `mono_atomic_cas_ptr` to make sure you do not leak memory here.

Reply:
This is a convergent leak, it's fine.

Reply:
I agree with @kumpera. on my system it's six bytes and anyways this case is pretty unlikely.